core/src/test/java/org/apache/iceberg/TestSchemaAndMappingUpdate.java

```diff
@@ -176,20 +176,11 @@ public void testModificationWithMetricsMetrics() {
               .set("write.metadata.metrics.column.ids", "full")
               .commit());
 
-    AssertHelpers.assertThrows(
-        "Deleting a column with metrics fails",
-        ValidationException.class,
-        null,
-        () ->
-          table.updateSchema()
-              .deleteColumn("id")
-              .commit());
+    // Re-naming a column with metrics succeeds;
+    table.updateSchema().renameColumn("id", "bloop").commit();
```

Comment:
I think we may also want to validate that the metric config is copied to the new name correctly, and deleted in the example below. Just to make sure the same value goes through

Reply:
Added :)